beacon_node/network/src/sync/block_sidecar_coupling.rs

```diff
@@ -300,82 +315,66 @@ impl<E: EthSpec> RangeBlockComponentsRequest<E> {
                 .insert(index, column)
                 .is_some()
             {
-                return Err(CouplingError {
-                    msg: format!("Repeated column block_root {block_root:?} index {index}"),
-                    column_and_peer: None,
-                });
+                return Err(CouplingError::InternalError(format!(
```

Comment:
Add a comment here saying that `DataColumnsByRangeRequestItems` ensures there are no duplicated indices and that peers return the requested indices, which we never request to more than one peer. Thus an internal error

Reply:
I removed the error here because we can still continue with duplicates, added a comment like you said though